browser/resources/settings/brave_clear_browsing_data_dialog/brave_clear_browsing_data_dialog_behavior.ts

```diff
@@ -1,151 +1,163 @@
-    super.ready();
+  private clearDataBrowserProxy_: BraveClearBrowsingDataDialogBrowserProxy =
+    BraveClearBrowsingDataDialogBrowserProxyImpl.getInstance()
+
+  private onSelectedPageChangedCallback_: (() => void) | null = null
+
+  private updateSaveButtonStateCallback_: (() => void) | null = null
 
-    // Append On exit tab to tab selector.
-    this.tabsNames_.push(loadTimeData.getString('onExitPageTitle'));
+  private saveOnExitSettingsCallback_: (() => void) | null = null
+
+  override ready() {
+    super.ready()
 
     this.addWebUiListener(
-      'update-counter-text', this.updateOnExitCountersText.bind(this));
+      'update-counter-text', this.updateOnExitCountersText.bind(this))
 
     this.addWebUiListener(
-      'brave-rewards-enabled-changed', (enabled: Boolean) => {
-      this.braveRewardsEnabled_ = enabled;
+      'brave-rewards-enabled-changed', (enabled: boolean) => {
+      this.braveRewardsEnabled_ = enabled
     })
 
     this.clearDataBrowserProxy_.getBraveRewardsEnabled().then((enabled) => {
-      this.braveRewardsEnabled_ = enabled;
+      this.braveRewardsEnabled_ = enabled
     })
   }
 
   override connectedCallback() {
-    super.connectedCallback();
+    super.connectedCallback()
 
-    this.onSelectedPageChangedCallback_ = this.onSelectedPageChanged_.bind(this);
+    this.onSelectedPageChangedCallback_ = this.onSelectedPageChanged_.bind(this)
     this.$.pages.addEventListener('iron-select',
-      this.onSelectedPageChangedCallback_);
+      this.onSelectedPageChangedCallback_)
 
-    this.updateSaveButtonStateCallback_ = this.updateSaveButtonState_.bind(this);
-    this.shadowRoot.querySelector('#on-exit-tab').addEventListener(
-      'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_);
+    this.updateSaveButtonStateCallback_ = this.updateSaveButtonState_.bind(this)
+    this.shadowRoot!.querySelector('#on-exit-tab')!.addEventListener(
+      'clear-data-on-exit-page-change', this.updateSaveButtonStateCallback_)
 
-    this.saveOnExitSettingsCallback_ = this.saveOnExitSettings_.bind(this);
-    this.shadowRoot.querySelector('#saveOnExitSettingsConfirm').addEventListener(
-      'click', this.saveOnExitSettingsCallback_);
+    this.saveOnExitSettingsCallback_ = this.saveOnExitSettings_.bind(this)
+    this.shadowRoot!.querySelector('#saveOnExitSettingsConfirm')!
+      .addEventListener('click', this.saveOnExitSettingsCallback_)
 
-    this.onClearBraveAdsDataClickHandler_ = this.clearBraveAdsData_.bind(this);
+    this.onClearBraveAdsDataClickHandler_ = this.clearBraveAdsData_.bind(this)
   }
 
   override disconnectedCallback() {
     super.disconnectedCallback()
 
     if (this.onSelectedPageChangedCallback_) {
       this.$.pages.removeEventListener('selected-item-changed',
-        this.onSelectedPageChangedCallback_);
-      this.onSelectedPageChangedCallback_ = null;
+        this.onSelectedPageChangedCallback_)
+      this.onSelectedPageChangedCallback_ = null
```

Comment:
if we use an arrow function or set up this handler in the constructor we won't need to null these out.

Reply:
Just letting this be for now, as this entire module clearly needs some additional refactoring that's a little beyond the scope of this PR.